extensions/elasticsearch-rest-high-level-client/runtime/src/main/java/io/quarkus/elasticsearch/restclient/highlevel/runtime/ElasticsearchRestHighLevelClientProducer.java

```diff
@@ -0,0 +1,41 @@
+package io.quarkus.elasticsearch.restclient.highlevel.runtime;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Collections;
+
+import javax.annotation.PreDestroy;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestHighLevelClient;
+
+@ApplicationScoped
+public class ElasticsearchRestHighLevelClientProducer {
+
+    @Inject
+    RestClient restClient;
```

Comment:
Just a heads-up: this will fail in an application that defines multiple rest clients (to communicate with multiple Elasticsearch clusters from the same application).

I know the current low-level REST client extension doesn't allow that, but users can still define their own beans (which is typically what I would do if I needed to communicate with multiple Elasticsearch clusters).

Reply:
I think we should add the `@Default` qualifier to the produced bean and the injection point here.